src/Microsoft.AspNetCore.OData/Routing/Conventions/ODataRoutingConventions.cs

```diff
@@ -38,6 +39,27 @@ public static class ODataRoutingConventions
             return routingConventions;
         }
 
+        /// <summary>
+        /// Creates a mutable list of the default OData routing conventions with attribute routing enabled.
+        /// </summary>
+        /// <param name="routeName">The name of the route.</param>
+        /// <param name="serviceProvider">The service provider.</param>
+        /// <returns>A mutable list of the default OData routing conventions.</returns>
+        public static IList<IODataRoutingConvention> CreateDefaultWithAttributeRouting(
+            string routeName,
+            IServiceProvider serviceProvider)
+        {
+            if (routeName == null)
+            {
+                throw Error.ArgumentNull("routeName");
+            }
+
+            IList<IODataRoutingConvention> routingConventions = CreateDefault();
```

Comment:
@xuzhg The 4 lines here are a replica of these others [here](https://github.com/OData/WebApi/pull/2035/files#diff-eb87a7b003f2e163e4936620def60725R36). Maybe the existing `CreateDefaultWithAttributeRouting` overload could call this new one like this `return CreateDefaultWithAttributeRouting(routeName, builder.ServiceProvider`) - in which case the check for null `routeName` wouldn't be required in the existing overload, or consider adding a helper private method that both overloads call

Reply:
:thumbsup:

---
In reply to: [384475452](https://github.com/OData/WebApi/pull/2035#discussion_r384475452) [](ancestors = 384475452)

Reply:
Changed.

Originally, i'd like to change the exiting public to the new one. However, it's a breaking change. So, i add a new one.
 

---
In reply to: [384715327](https://github.com/OData/WebApi/pull/2035#discussion_r384715327) [](ancestors = 384715327,384475452)